aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,240 @@
+        else:
+            raise exceptions.InputValidationError(
+                "Unsupported value, {}, for 'plot_number' ('iflag' in pp.x docs)".format(dimension)
+            )
+
+        # =================== NAMELISTS AND CARDS ========================
+        try:
+            namelists_toprint = settings.pop('NAMELISTS')
+            if not isinstance(namelists_toprint, list):
+                raise exceptions.InputValidationError(
+                    "The 'NAMELISTS' value, if specified in the settings input node, must be a list of strings")
+        except KeyError:  # list of namelists not specified; do automatic detection
+            namelists_toprint = self._default_namelists
+
+        input_filename = self.inputs.metadata.options.input_filename
+        with folder.open(input_filename, 'w') as infile:
+            for namelist_name in namelists_toprint:
+                infile.write(u'&{0}\n'.format(namelist_name))
+                # namelist content; set to {} if not present, so that we leave an empty namelist
+                namelist = parameters.pop(namelist_name, {})
+                for key, value in sorted(six.iteritems(namelist)):
+                    infile.write(convert_input_to_namelist_entry(key, value))
+                infile.write(u'/\n')
+
+
+        # Check for specified namelists that are not expected
+        if parameters:
+            raise exceptions.InputValidationError(
+                'The following namelists are specified in parameters, but are '
+                'not valid namelists for the current type of calculation: '
+                '{}'.format(','.join(list(parameters.keys()))))
+
+        remote_copy_list = []
+        local_copy_list = []
+
+        # copy remote output dir, if specified
+        parent_calc_folder = self.inputs.get('parent_folder', None)
+        if parent_calc_folder is not None:
+            if isinstance(parent_calc_folder, orm.RemoteData):
+                parent_calc_out_subfolder = settings.pop('PARENT_CALC_OUT_SUBFOLDER', self._INPUT_SUBFOLDER)
+                remote_copy_list.append((
+                    parent_calc_folder.computer.uuid,
+                    os.path.join(parent_calc_folder.get_remote_path(), parent_calc_out_subfolder),
+                    self._OUTPUT_SUBFOLDER
+                ))
+            elif isinstance(parent_calc_folder, orm.FolderData):
+                for filename in parent_calc_folder.list_object_names():
+                    local_copy_list.append((
+                        parent_calc_folder.uuid,
+                        filename,
+                        os.path.join(self._OUTPUT_SUBFOLDER, filename)
+                    ))
+            elif isinstance(parent_calc_folder, orm.SinglefileData):
+                single_file = parent_calc_folder
+                local_copy_list.append(
+                    (single_file.uuid, single_file.filename, single_file.filename)
+                )
+
```

Comment:
according to your spec this input cannot be a `SingleFileData`

Reply:
Indeed it cannot. This was a copy-pasted stanza that I'll remove. Well spotted.